### ISSUE TITLE
Hiding classlist in DoxygenLayout.xml causes the C structures to be shown in parent hierarchy level in html navigation tree

### ISSUE THREAD

**Issue by ChristianHerget** [doxygen/doxygen#8328]
The following has been observed with v1.9.0 and v1.9.1 (x64 windows).

----

Please consider the following xml fragment
```
    <tab type="classes" visible="yes" title="">
      <tab type="classlist" visible="no" title="" intro=""/>
      <tab type="classindex" visible="no" title=""/> 
      <tab type="hierarchy" visible="no" title="" intro=""/>
      <tab type="classmembers" visible="no" title="" intro=""/>
    </tab>
```

This will causes the list of C structures to be shown on the level of _classes_ (Data Structures) in the html navigation tree (CHM Contents). 
The structs in the navigation tree will disappear when using v1.8.20.

----

Please consider the following xml fragment
```
    <tab type="classes" visible="no" title="">
      <tab type="classlist" visible="no" title="" intro=""/>
      <tab type="classindex" visible="no" title=""/> 
      <tab type="hierarchy" visible="no" title="" intro=""/>
      <tab type="classmembers" visible="no" title="" intro=""/>
    </tab>
```

This will causes the list of C structures to be shown on the top level in the html navigation tree (CHM Contents). 
The structs in the navigation tree will disappear when using v1.8.20.

----

Example project:

[Issue.zip](https://github.com/doxygen/doxygen/files/5803130/Issue.zip)

Example output with v1.8.20:

![image](https://user-images.githubusercontent.com/11754024/104336010-3bfb2400-54f4-11eb-8121-cf1de5b8eb3d.png)

Example output with v1.9.1:

![image](https://user-images.githubusercontent.com/11754024/104336025-41586e80-54f4-11eb-99b4-52b4ffebbbd0.png)

**Comment by albert-github** [comment]
- Can you please attach a, small, self contained example (source+configuration file in a tar or zip) that allows us to reproduce the problem? Please don't add external links as they might not be persistent.

**Comment by ChristianHerget** [comment]
> 
> 
>     * Can you please attach a, small, self contained example (source+configuration file in a tar or zip) that allows us to reproduce the problem? Please don't add external links as they might not be persistent.

I attached a small example and updated my original comment accordingly.

**Comment by albert-github** [comment]
You explicitly set to `OPTIMIZE_OUTPUT_FOR_C  = YES` so the C rules should adhere and not the Cpp rules.
In Cpp a struct and a class are a bit similar and hence doxygen probably got confused.

**Comment by albert-github** [comment]
As far as I can see the commit:
```
Commit: ade26d54258e932274b2605bdc67ca45af4b398a [ade26d5]
Date: Monday, October 19, 2020 1:31:00 PM
Merge pull request #8112 from albert-github/feature/bug_ext_map_msg

Confusing message for extension mapping for extension with starting dot.
```
was still OK, but the next commit has the problem:
```
Commit: 33b0f4d25dff25b0e50d62eff68155106e88d58d [33b0f4d]
Date: Monday, October 19, 2020 4:03:07 PM
issue #8103: C++ Table of content, namespace list does not contains namespace without class (part 2)
```

**Comment by DGA45** [comment]
I have the same issue (ANSI-C source code), and this is polluting our generated documentation.
Any workaround or news for a fix ?
Thanks and regards

**Comment by ChristianHerget** [comment]
It seems that this issue got fixed with v1.9.2. Thanks for this.

**Comment by albert-github** [comment]
Indeed it looks like everything is indeed working as intended.

I think that the problem has been solved in the mean time.
